app/services/messaging/twilio/otp_sms.rb

```diff
@@ -0,0 +1,16 @@
+class Messaging::Twilio::OtpSms < Messaging::Twilio::Api
```

Comment:
Why a separate service for OTP SMS? More specifically, why are we tying purpose to these classes? I thought we would make these classes strictly _channels_ for any message.

Can we just have an SMS class? Nothing wrong in the OTP SMS's also leveraging an array of senders, right?

Reply:
The idea is to surface all the options available to send out a message transparently to the caller. Note that the purpose in the channel's name (Reminder, Otp) is not the purpose of the message, but the purpose of the channel itself. On our twilio dashboard we have two "services", in twilio parlance: The `Appointment Reminder Senders` pool and the one number we use for OTPs. The OTP sender exists to send OTPs. The reminder pool exists to send reminders. These channels mirror that setup.

If we wanted to send OTP SMSes via a list of senders, we would do one of the following:
- Use the same senders as the reminder senders. The OTP job should simply switch to calling `ReminderSms` in that case. This is a hack though, and dealing with it will not be the SMS class's concern. That complexity should be on the caller.
- We add more senders dedicated to OTP, in which case `OtpSms` deals with picking one of them.

Combining the OTP and reminder channels also makes the interface non-generic, where the caller would select the effective channel by passing in a flag. The SMS implementation would then try to pick a channel based on a `case` statement. This bundling doesn't buy us anything since all the SMS code is anyways shared nicely in the other implementation.

Reply:
I was gonna say the OTP and reminder messages don't have to choose diffferent senders in Twilio, but I remember that one advantage to separation was that any blockage or overage in our reminder messaging (rate limits etc) wouldn't affect OTPs. In that case, yeah, this looks great.